lifecycle.go

```diff
@@ -98,7 +98,7 @@ func (l *lifecycle) stop() error {
 }
 
 // NewTestLifecycle creates a new test lifecycle
-func NewTestLifecycle() Lifecycle {
+func NewTestLifecycle() *TestLifecycle {
```

Comment:
Looks ok, could you add a compile time check, that TestLifecycle implements Lifecycle?